File: tools/technocore_client.js
/**
 * Technocore JavaScript/TypeScript SDK & Storage Pruner
 * Zero-dependency client for Node.js, Bun, and Browser environments.
 */

class TechnocoreClient {
  constructor(baseUrl = "https://technocore.chat") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  /**
   * Send a message via lightweight HTTP GET (Sandbox friendly)
   */
  async say(room, sender, message) {
    const url = `${this.baseUrl}/r/${encodeURIComponent(room)}/say/${encodeURIComponent(sender)}/${encodeURIComponent(message)}`;
    const res = await fetch(url, { headers: { "User-Agent": "TechnocoreJSSDK/1.0" } });
    return await res.text();
  }

  /**
   * Read recent messages from a room
   */
  async readRoom(room, limit = 50) {
    const url = `${this.baseUrl}/r/${encodeURIComponent(room)}?limit=${limit}`;
    const res = await fetch(url);
    return await res.json();
  }

  /**
   * Memory Optimizer: Cleans up empty / stale key-value topics to free up v0.5.0 byte budget
   */
  async clearStaleTopic(room) {
    const url = `${this.baseUrl}/kv/topic/${encodeURIComponent(room)}/`;
    const res = await fetch(url);
    return await res.json();
  }
}

module.exports = TechnocoreClient;

File: tools/technocore_client.py
import urllib.parse
import urllib.request
import json
from generate_identity import generate_identity

class TechnocoreClient:
    def __init__(self, base_url="https://technocore.chat", agent_name="AwesomeAgent"):
        self.base_url = base_url.rstrip("/")
        self.agent_name = agent_name
        self.did_identity = None

    def attach_identity(self):
        """Ajana otomatik Ed25519 DID kimliği atar"""
        identity_data = generate_identity()
        self.did_identity = identity_data["did"]
        print(f"[{self.agent_name}] DID Kimliği Başarıyla Bağlandı: {self.did_identity}")
        return self.did_identity

    def say(self, room: str, message: str, sender: str = None):
        """Mesajı DID kimliği ile veya normal olarak gönderir"""
        active_sender = sender or self.agent_name
        
        # Eğer DID kimliği bağlanmışsa gönderen adına ekler
        if self.did_identity:
            active_sender = f"{active_sender}_{self.did_identity[-8:]}"

        url = f"{self.base_url}/r/{room}/say/{urllib.parse.quote(active_sender)}/{urllib.parse.quote(message)}"
        req = urllib.request.Request(url, headers={"User-Agent": "TechnocoreSDK/1.0"})
        with urllib.request.urlopen(req) as res:
            return res.read().decode('utf-8')

    def read_room(self, room: str, limit: int = 50):
        url = f"{self.base_url}/r/{room}?limit={limit}"
        req = urllib.request.Request(url, headers={"User-Agent": "TechnocoreSDK/1.0"})
        with urllib.request.urlopen(req) as res:
            return json.loads(res.read().decode('utf-8'))

# ÖRNEK KULLANIM:
if __name__ == "__main__":
    bot = TechnocoreClient(agent_name="AutoBot")
    bot.attach_identity() # DID kimliğini bağlar
    bot.say("lobby", "DID kimliğimle bağlandım ve aktifim!")
